auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/VirtualGroupService.java

```diff
@@ -0,0 +1,71 @@
+package com.sequenceiq.cloudbreak.auth.altus;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+import com.sequenceiq.cloudbreak.auth.security.InternalCrnBuilder;
+
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+
+@Component
+public class VirtualGroupService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(VirtualGroupService.class);
+
+    private static final String IAM_INTERNAL_ACTOR_CRN = new InternalCrnBuilder(Crn.Service.IAM).getInternalCrnForServiceAsString();
+
+    @Inject
+    private GrpcUmsClient grpcUmsClient;
+
+    public Map<UmsRight, String> createVirtualGroups(String accountId, String environmentCrn) {
+        Map<UmsRight, String> virtualGroups = new HashMap<>();
+        for (UmsRight right : UmsRight.values()) {
+            virtualGroups.put(right, createOrGetVirtualGroup(accountId, environmentCrn, right.getRight()));
+        }
+        return virtualGroups;
+    }
+
+    public String getVirtualGroup(String adminGroup, String accountId, String environmentCrn, String right) {
```

Comment:
We could create a new method which takes the VirtualGroupView as parameter and we can reduce the code duplications for each invocation with the same parameters
```
virtualGroupService.getVirtualGroup(virtualGroupView.getAdminGroup(), virtualGroupView.getAccountId(),
                            virtualGroupView.getEnvironmentCrn(),...
```